channeldb/payments.go

```diff
@@ -317,6 +344,124 @@ func deserializeOutgoingPayment(r io.Reader) (*OutgoingPayment, error) {
 	return p, nil
 }
 
+// CreationInfo is the information necessary to have ready when initiating a
+// payment, moving it into state InFlight.
+type CreationInfo struct {
+	// PaymentHash is the hash this payment is paying to.
+	PaymentHash lntypes.Hash
+
+	// Value is the amount we are paying.
+	Value lnwire.MilliSatoshi
+
+	// CreatingDate is the time when this payment was initiated.
+	CreationDate time.Time
+
+	// PaymentRequest is the full payment request, if any.
+	PaymentRequest []byte
+}
+
+// AttemptInfo contains information about a specific payment attempt for a
+// given payment. This information is used by the router to handle any errors
+// coming back after an attempt is made, and to query the switch about the
+// status of a payment. For settled payment this will be the information for
+// the succeeding payment attempt.
+type AttemptInfo struct {
+	// PaymentID is the unique ID used for this attempt.
+	PaymentID uint64
+
+	// Route is the route attempted to send the HTLC.
+	Route route.Route
```

Comment:
Nice, this can also eventually be used to analysis to determine which outgoing channels we favor, which we can use to eventually make channel pruning decisions.